wetterdienst/provider/dwd/observation/fileindex.py

```diff
@@ -53,23 +56,28 @@ def create_file_list_for_climate_observations(
 
 
 def create_file_index_for_climate_observations(
-    parameter_set: DwdObservationDataset,
+    dataset: DwdObservationDataset,
     resolution: Resolution,
     period: Period,
 ) -> pd.DataFrame:
     """
     Function (cached) to create a file index of the DWD station data. The file index
     is created for an individual set of parameters.
     Args:
-        parameter_set: parameter of Parameter enumeration
+        dataset: parameter of Parameter enumeration
         resolution: time resolution of TimeResolution enumeration
         period: period type of PeriodType enumeration
     Returns:
         file index in a pandas.DataFrame with sets of parameters and station id
     """
     timezone_germany = timezone("Europe/Berlin")
 
-    file_index = _create_file_index_for_dwd_server(parameter_set, resolution, period, DWDCDCBase.CLIMATE_OBSERVATIONS)
+    if dataset in DWD_URBAN_DATASETS:
```

Comment:
You can create a dict with a mapping between dataset objects and the Base classes DWDCDCBase etc. Then you do not have to introduce  if/else clause here.